backend/src/main/java/de/bund/digitalservice/ris/norms/application/service/ArticleService.java

```diff
@@ -28,27 +33,101 @@ public ArticleService(
   }
 
   @Override
-  public Optional<String> loadArticleHtml(final Query query) {
-    return loadNormPort
-        .loadNorm(new LoadNormPort.Command(query.eli()))
-        .map(
-            norm -> {
-              if (query.atIsoDate() == null) return norm; // no date given -> use the norm unchanged
-              else {
-                return timeMachineService.applyPassiveModifications(
-                    new ApplyPassiveModificationsUseCase.Query(norm, query.atIsoDate()));
-              }
-            })
-        .map(Norm::getArticles)
-        .stream()
-        .flatMap(List::stream)
+  public String loadArticleHtml(final LoadArticleHtmlUseCase.Query query) {
+    var norm =
+        loadNormPort
+            .loadNorm(new LoadNormPort.Command(query.eli()))
+            .orElseThrow(() -> new NormNotFoundException(query.eli()));
+
+    if (query.atIsoDate() != null) {
+      norm =
+          timeMachineService.applyPassiveModifications(
+              new ApplyPassiveModificationsUseCase.Query(norm, query.atIsoDate()));
+    }
+
+    return norm.getArticles().stream()
         .filter(
             article -> article.getEid().isPresent() && article.getEid().get().equals(query.eid()))
         .findFirst()
         .map(article -> XmlMapper.toString(article.getNode()))
         .map(
             xml ->
                 xsltTransformationService.transformLegalDocMlToHtml(
-                    new TransformLegalDocMlToHtmlUseCase.Query(xml, false, false)));
+                    new TransformLegalDocMlToHtmlUseCase.Query(xml, false, false)))
+        .orElseThrow(() -> new ArticleNotFoundException(query.eli(), query.eid()));
+  }
+
+  @Override
+  public List<Article> loadArticlesFromNorm(final LoadArticlesFromNormUseCase.Query query) {
+    final var amendedAt = query.amendedAt();
+    final var amendedBy = query.amendedBy();
+
+    final var norm =
+        loadNormPort
+            .loadNorm(new LoadNormPort.Command(query.eli()))
+            .orElseThrow(() -> new NormNotFoundException(query.eli()));
+
+    List<Article> articles = norm.getArticles();
+
+    // Filter list of articles by passive mods if at least one filter is specified
+    if (amendedBy != null || amendedAt != null) {
+      var filterPassiveMods = getPassiveModsAmendingByOrAt(norm, amendedBy, amendedAt);
```

Comment:
I get your encapsulation to have smaller methods by for me it was more confusing because of the creation if a Predicate<> applied then in the filter...

I would just have 1 big complicated method but maybe a bit more structured, like:

```
    // Filter list of articles by passive mods if at least one filter is specified
    if (amendedBy != null || amendedAt != null) {
      return filterArticlesByAmendedByOrAt(norm, amendedBy, amendedAt);
    }
```
    
    With the big method being:
    
```
private List<Article> filterArticlesByAmendedByOrAt(final Norm fromNorm, final String amendedBy, final String amendedAt) {
    if (amendedBy == null && amendedAt == null) return List.of();

    final List<TextualMod> passiveMods = fromNorm.getMeta()
            .getAnalysis()
            .map(Analysis::getPassiveModifications)
            .orElse(Collections.emptyList());

    final List<TextualMod> filteredPassiveMods = passiveMods.stream()
            .filter(passiveModification -> {
              if (amendedBy != null) {
                return passiveModification.getSourceHref()
                        .flatMap(Href::getEli)
                        .map(amendedBy::equals)
                        .orElse(false);
              }
              return true;
            })
            .filter(passiveModification -> {
              if (amendedAt != null) {
                return passiveModification.getForcePeriodEid()
                        .flatMap(fromNorm::getStartEventRefForTemporalGroup)
                        .map(amendedAt::equals)
                        .orElse(false);
              }
              return true;
            })
            .toList();

    return fromNorm.getArticles().stream().filter(article ->  // If we filter by amendedAt or amendedBy: Those properties are found
            // in the passive modifications we already collected above. What's left
            // now is to only return the articles that are going to be modified by
            // those passive modifications.
            filteredPassiveMods.stream()
                    .map(TextualMod::getDestinationHref)
                    .flatMap(Optional::stream)
                    .map(Href::getEId)
                    .flatMap(Optional::stream)
                    .anyMatch(
                            destinationEid ->
                                    // Modifications can be either on the article itself or anywhere
                                    // inside the article, hence the "contains" rather than exact
                                    // matching.
                                    destinationEid.contains(article.getEid().orElseThrow()))).toList();

  }
```

Reply:
As discussed—will keep it like this